src/cryptsetup/cryptsetup.c

```diff
@@ -497,16 +496,7 @@ static int attach_luks_or_plain(struct crypt_device *cd,
         assert(name);
         assert(key_file || passwords);
 
-        if (!arg_type || STR_IN_SET(arg_type, ANY_LUKS, CRYPT_LUKS1)) {
```

Comment:
hmm, is there a libcryptsetup call that tells us whether a superblock was already loaded? maybe that'd be a nicer approach here? i.e. load the super block only when it's not loaded yet?

Reply:
Not every crypt type has superblock (plain, loopaes), but for LUKS the crypt_get_type() can be used (as below).

(Just note: Truecrypt/Veracrypt has a superblock, but encrypted, so you need to have password to decrypti. LUKS has visible superblock that can be loaded without the password, so we have alwatys two possible paths for crypt_load() ...)

Reply:
well, superblock or not, what i was basically saying is whether it wouldn't be better to have some call "are settings in place already?" regardless where they come from. So that we configur things once, but not twice...

Reply:
I think crypt_get_type() does what you ask for. It returns NULL in case 'cd' context is empty (not initialized). Iow it returns non empty type only if context is initialized properly to some type and therefore you know calling crypt_load is not needed.